### PR TITLE
Improve lit testing scripts.

### PR DESCRIPTION
**Pull request by tsymalla**
- Let `check-llvm-dialects-lit` depend on the `llvm-dialects-test-depends` custom target
- Convert `add_dialects_unit_test` to a macro

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -42,10 +42,9 @@ configure_lit_site_cfg(
 
 add_lit_testsuite(check-llvm-dialects-lit "Running the llvm-dialects regression tests"
   ${CMAKE_CURRENT_BINARY_DIR}
-  ${exclude_from_check_all}
-  DEPENDS ${LLVM_DIALECTS_TEST_DEPENDS}
+  ${EXCLUDE_FROM_CHECK_ALL}
+  DEPENDS llvm-dialects-test-depends
 )
-set_target_properties(check-llvm-dialects-lit PROPERTIES FOLDER "Tests")
```
Comment (nhaehnle) [inline test/CMakeLists.txt:48]: Why did you remove this? AFAIK it is relevant for Visual Studio.

**Review by nhaehnle — COMMENTED** [review]
Not sure I understand what all the changes are supposed to do, but it seems reasonable enough overall.

File: test/CMakeLists.txt (hunk 1 above)
Comment (tsymalla-AMD) [reply test/CMakeLists.txt:48]: It is already set on llvm-dialects-test-depends (not sure if it works though, going to double-check)
